app/models/dossier.rb

```diff
@@ -627,4 +641,35 @@ def send_web_hook
       )
     end
   end
+
+  def self.send_brouillon_expiration_notices
+    brouillons = Dossier
+      .brouillon_close_to_expiration
+      .without_notice_sent
+
+    brouillons
+      .includes(:user)
+      .group_by(&:user)
+      .each do |(user, dossiers)|
+        DossierMailer.notify_brouillon_near_deletion(user, dossiers).deliver_later
+      end
+
+    brouillons.update_all(brouillon_close_to_expiration_notice_sent_at: Time.zone.now)
+  end
+
+  def self.destroy_brouillons_and_notify
+    expired_brouillons = Dossier.expired_brouillon
+
+    expired_brouillons
+      .includes(:procedure, :user)
+      .group_by(&:user)
+      .each do |(user, dossiers)|
+
+      dossier_hashes = dossiers.map(&:hash_for_deletion_mail)
+      DossierMailer.notify_brouillon_deletion(user, dossier_hashes).deliver_later
+      dossiers.each { |d| DeletedDossier.create_from_dossier(d) }
+    end
+
+    expired_brouillons.destroy_all
```

Comment:
Ça m'inquiète légèrement qu'on ne destroy qu'à la fin de la boucle : si un des dossiers plante, l'étape suivante du job va renvoyer l'email de notif, encore et encore.

Est-ce que ça ne serait pas plus sûr (même si légèrement moins performant) de faire le destroy à chaque itération de la boucle ?